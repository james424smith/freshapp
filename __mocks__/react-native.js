import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
