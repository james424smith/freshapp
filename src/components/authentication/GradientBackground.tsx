import { useTheme } from "@react-navigation/native";
import React from "react";
import { View } from "react-native";
import LinearGradient from "react-native-linear-gradient";
import styles from "../../styles/AuthStyles";

type Props = {
  children: JSX.Element;
};

const GradientBackground = (props: Props) => {
  const theme = useTheme();

  if (theme.dark) {
    return (
      <View
        style={{
          ...styles.signUpContainer,
          backgroundColor: theme.colors.signInPageBackground,
        }}
      >
        {props.children}
      </View>
    );
  } else {
    return (
      <LinearGradient
        start={{ x: 0.1, y: 0 }}
        end={{ x: 0, y: 1.5 }}
        locations={[0, 0.5]}
        style={styles.signUpContainer}
        colors={["#099BBD", "#005AA5"]}
      >
        {props.children}
      </LinearGradient>
    );
  }
};

export default GradientBackground;
