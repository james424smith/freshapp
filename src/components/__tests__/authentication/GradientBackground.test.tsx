import React from "react";
import GradientBackground from "../../authentication/GradientBackground";
import { render, cleanup } from "@testing-library/react-native";
import { View } from "react-native";
import * as RNA from "react-native-appearance";

describe("BackIcon snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui when scheme is dark", () => {
    Object.defineProperty(
      RNA,
      "useColorScheme",
      jest.fn().mockReturnValue("dark")
    );
    const { toJSON } = render(
      <GradientBackground>
        <View />
      </GradientBackground>
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when scheme is light", () => {
    Object.defineProperty(
      RNA,
      "useColorScheme",
      jest.fn().mockReturnValue("light")
    );
    const { toJSON } = render(
      <GradientBackground>
        <View />
      </GradientBackground>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
