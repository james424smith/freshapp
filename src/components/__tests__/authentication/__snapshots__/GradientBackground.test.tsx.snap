// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon snapshot test should render correct ui when scheme is dark 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278819773,
      4278213285,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 1.5,
    }
  }
  locations={
    Array [
      0,
      0.5,
    ]
  }
  startPoint={
    Object {
      "x": 0.1,
      "y": 0,
    }
  }
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "width": 750,
    }
  }
>
  <View />
</BVLinearGradient>
`;

exports[`BackIcon snapshot test should render correct ui when scheme is light 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278819773,
      4278213285,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 1.5,
    }
  }
  locations={
    Array [
      0,
      0.5,
    ]
  }
  startPoint={
    Object {
      "x": 0.1,
      "y": 0,
    }
  }
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "width": 750,
    }
  }
>
  <View />
</BVLinearGradient>
`;
